source/System.cpp

```diff
@@ -264,6 +264,69 @@ void System::Load(const DataNode &node, Set<Planet> &planets)
 				belts.emplace_back(weight, radius);
 			}
 		}
+		else if(key == "object")
+		{
+			if(remove)
+			{
+				StellarObject toRemoveTemplate;
+
+				for(const DataNode &grand : child)
+				{
+					if(grand.Token(0) == "sprite" && grand.Size() >= 2)
+						toRemoveTemplate.LoadSprite(grand);
+					else if(grand.Token(0) == "distance" && grand.Size() >= 2)
+						toRemoveTemplate.distance = grand.Value(1);
+					else if(grand.Token(0) == "period" && grand.Size() >= 2)
+						toRemoveTemplate.speed = 360. / grand.Value(1);
+					else if(grand.Token(0) == "offset" && grand.Size() >= 2)
+						toRemoveTemplate.offset = grand.Value(1);
+					else if(grand.Token(0) == "hazard" || grand.Token(0) == "object")
+						grand.PrintTrace("Invalid child node when removing object:");
+					else
+						grand.PrintTrace("Skipping unrecognized attribute:");
+				}
```

Comment:
Maybe have a separate function that only loads objects (with an input bool that says if it's adding or removing so that it can skip over fields it doesn't need to worry about on removing) so that this bit isn't mostly duplicated from LoadObject.